packages/manager/modules/billing/src/components/services-actions/services-actions.html

```diff
@@ -100,7 +100,12 @@
         data-ng-if="(!$ctrl.service.shouldDeleteAtExpiration() || !$ctrl.service.isResiliated()) && !$ctrl.service.hasDebt() && !$ctrl.service.hasPendingResiliation()"
     >
         <oui-action-menu-item
-            ng-if="$ctrl.service.canBeResiliated($ctrl.user.nichandle)"
+            ng-if="$ctrl.service.canBeResiliated($ctrl.user.nichandle) && !$ctrl.service.engagementDetails && !$ctrl.service.canBeEngaged
+            || (
+              $ctrl.service.engagementDetails
+              && $ctrl.service.engagementDetails.endRule.strategy === 'REACTIVATE_ENGAGEMENT'
+              && $ctrl.service.engagementDetails.endRule.possibleStrategies.length > 0
```

Comment:
`service` has a method `isEngaged` that we could reuse. Also we could maybe abstract those conditions by having method in the class used. For instance `canReacrivateEngagement()` or `canResiliateEngagement`

Reply:
Here `service` is not a instance of `Service` class, but is one of `BillingService` class.
So this won't be able to be used. But method could be done in the component controller, which is not very good.
What do you think ?

Reply:
Is there any issue adding those to the `BillingService` class then ?

Reply:
Hmm I find it confusing that `BillingService`class handle these information as they are not part of what is returned by `/serviceInfos`.

Reply:
Why add it here then https://github.com/ovh/manager/pull/4189/files#diff-eb74d56185f345b0fdc694c3333879f25a6725b2958587f4da1e77543edb3e43R76 ? 😅 
Here it works fine because we don't destructure the properties in the constructor, if it's confusing then we should find another way to pass this information than through the created instance

Reply:
TBH, it is more a "copy" of what you've done here: https://github.com/ovh/manager/pull/3992/files#diff-eb74d56185f345b0fdc694c3333879f25a6725b2958587f4da1e77543edb3e43R63-R64. 

I agree this is not the best though.